Document agent detailed upgrade states (#596)
elastic/ingest-docs: docs/en/ingest-management/fleet/upgrade-elastic-agent.asciidoc

[[view-upgrade-status]]
== View upgrade status

On the **Agents** tab, when you trigger an upgrade, agents that are currently 
upgrading show a detailed upgrade status indicator beside the agent version.

[role="screenshot"]
image::images/upgrade-states.png[Detailed state of an upgrading agent]

The following table explains the various possible upgrade states.

.{agent} upgrade states
|===
| State | Description

> | Upgrade downloading | {agent} is downloading the archive containing the new version artifact.

Reviewer: How about listing the upgrade states in the order they happen? These can be found [here](https://github.com/elastic/elastic-agent/issues/3119).

| Upgrade extracting | {agent} is extracting the new version artifact from the downloaded archive.
| Upgrade failed | An error has been detected in the newly upgraded {agent} and the attempt to roll the upgrade back to the previous version has failed.
| Upgrade monitoring | The newly upgraded {agent} has started as is being monitored for errors.
| Upgrade replacing | {agent} is currently replacing the former, pre-upgrade agent artifact with the new one.
| Upgrade requested | {agent} has received the upgrade notice from {fleet} and the upgrade will start when the request has been received.
| Upgrade restarting | {agent} has been replaced with a new version and is now restarting in order to apply the update.
| Upgrade rolled Back | Due to an error detected in the newly upgraded {agent}, it has been rolled back to the previous version.
| Upgrade scheduled | {agent} has received the upgrade notice from {fleet} and the upgrade will start at the indicated time.

|===

> As well, for {agents} on version 8.11 and higher, you can hover your cursor over the information icon to get more granular details about the upgrade state.

Reviewer: The target version is currently 8.12, although that might change.

[[view-upgrade-status]]
== View upgrade status

On the Agents tab, when you trigger an upgrade, agents that are upgrading have the status `Updating` until the upgrade is complete, and then the status changes back to `Healthy`.

Agents on version 8.12 and higher that are currently upgrading additionally show a detailed upgrade status indicator.

[role="screenshot"]
image::images/upgrade-states.png[Detailed state of an upgrading agent]

The following table explains the upgrade states in the order that they can occur.

.{agent} upgrade states
|===
| State | Description

> | Upgrade requested | {agent} has received the upgrade notice from {fleet} and the upgrade will start when the request has been received.

Reviewer 2: I'd omit the second clause in the description because it isn't accurate for scheduled upgrades.

```suggestion
| Upgrade requested | {agent} has received the upgrade notice from {fleet}.
```

| Upgrade scheduled | {agent} has received the upgrade notice from {fleet} and the upgrade will start at the indicated time.
| Upgrade downloading | {agent} is downloading the archive containing the new version artifact.
| Upgrade extracting | {agent} is extracting the new version artifact from the downloaded archive.
| Upgrade replacing | {agent} is currently replacing the former, pre-upgrade agent artifact with the new one.
| Upgrade restarting | {agent} has been replaced with a new version and is now restarting in order to apply the update.

> | Upgrade monitoring | The newly upgraded {agent} has started as is being monitored for errors.

Reviewer 2: Typo?

```suggestion
| Upgrade monitoring | The newly upgraded {agent} has started and is being monitored for errors.
```

> | Upgrade rolled back | The upgrade was unsuccessful. {agent} is being rolled back to a previous version.

Reviewer 2: Consistency with terms used in the "Upgrade replacing" description.

```suggestion
| Upgrade rolled back | The upgrade was unsuccessful. {agent} is being rolled back to the former, pre-upgrade version.
```

| Upgrade failed | An error has been detected in the newly upgraded {agent} and the attempt to roll the upgrade back to the previous version has failed.

|===

Beside the upgrade status indicator, you can hover your cursor over the information icon to get more detail about the upgrade.

[role="screenshot"]
image::images/upgrade-detailed-state01.png[Granular upgrade details shown as hover text (agent has requested an upgrade)]

[role="screenshot"]
image::images/upgrade-detailed-state02.png[Granular upgrade details shown as hover text (agent is restarting to apply the update)]

> Note that when you upgrade agents on versions below 8.12, the upgrade details are not provided.

Reviewer 2: Perhaps less ambiguous?

```suggestion
Note that when you upgrade agents from versions below 8.12, the upgrade details are not provided.
```
